src/citrine/resources/design_workflow.py

```diff
@@ -16,13 +17,65 @@ class DesignWorkflowCollection(Collection[DesignWorkflow]):
     _individual_key = None
     _collection_key = 'response'
     _resource = DesignWorkflow
+    _api_version = "v2"
 
-    def __init__(self, project_id: UUID, session: Session):
+    def __init__(self, project_id: UUID, session: Session, branch_id: UUID = None):
         self.project_id: UUID = project_id
         self.session: Session = session
+        self.branch_id: UUID = branch_id
+
+    def register(self, model: DesignWorkflow) -> DesignWorkflow:
+        """
+        Upload a new design workflow.
+
+        The model's branch ID is ignored in favor of this collection's. If this
+        collection has a null branch ID, then the model is uploaded to the v1
+        endpoint and retrieved separately, to ensure all details are included.
+
+        Parameters
+        ----------
+        model: DesignWorkflow
+            The design workflow to be uploaded.
+
+        Return
+        -------
+        DesignWorkflow
+            The newly created design workflow.
+
+        """
+        if self.branch_id is None:
```

Comment:
in this case, might it be worthwhile to check that predictor and design space aren't None?

Reply:
I was on the fence, but I think I agree with you.